### DISCUSSION TITLE
How to assign the same random order of scenarios (pages) to different subgroups of participants

### DISCUSSION THREAD

**Discussion by markusgermar** [ctreffe/alfred#227] [Alfred Q&A]
Hi there!

I would like to present a number of scenarios (i.e., pages) in random order. Furthermore, I would like that all participants entering the same number (i.e., their group number) before being presented with the scenarios receive the same random order. For example, all participants entering the number "3" (because they are all in the group "3") see the scenarios in the same random order. Since I don't know how many groups will enter the study and which number they will use, I do not want to pre-generate a list of different orders. Additionally, the order of scenarios (pages) should be saved in a variable.

Can you help me? I think I have to use some random seed function, but I am not sure.

Thank you!

**Comment by ctreffe** [comment]
Indeed, you can set the same random order for multiple participants in a non-interactive experiment by setting the random seed to the same constant for each participant, as shown in the following pseudo code example:

```python
import random

random.seed(group_number)

random.shuffle(scenario_pages)

random.seed()
```

__ATTENTION__: It is important to reset the random seed to its default state after shuffeling the scenarios to prevent unwanted side effects from the constant seed. 

To save the page order (assuming you create the pages in a _for loop_), you can use alfred's [Value element](https://alfredo3.psych.bio.uni-goettingen.de/docs/generated/alfred3.element.misc.Value.html#alfred3.element.misc.Value) and a simple loop counter. Since you want to use a number entered by your participants as your randomization seed, I would recommend you add an empty section (_ScenarioSection_ in the following example) at the appropriate place in your experiment (where you want the scenarios in their randomized order to be presented), which you can access and fill from the _on\_first\_hide()_ method of any page (preferably the one with the group number data at the beginning of your experiment):

```python
# This code belongs in an on_first_hide() method

# Step 1 - randomization with fixed seed

random.seed(group_number)

random.shuffle(scenario_pages)

random.seed()

# Step 2 - add pages in randomized order to an existing section

scen_section = self.exp.all_sections.get('ScenarioSection')  # Find an existing section in the experiment

for counter, page in enumerate(scenario_pages):

    page += al.Value(counter+1, name="position")   # Adds a variable containing the page order

    scen_section += page  # Adds the page to an existing section in the experiment

```

**Reply by markusgermar** [reply]
Thank you!
